source/extensions/filters/network/zookeeper_proxy/decoder.cc

```diff
@@ -658,6 +661,10 @@ std::chrono::milliseconds DecoderImpl::fetchControlRequestData(const int32_t xid
   }
 
   std::queue<RequestBegin> rq_queue = it->second;
```

Comment:
Sorry for missing this. We should use reference here and the tests didn't catch this. Might want different `RequestBegin` content in the test.